src/actions/ast.js

```diff
@@ -126,3 +138,64 @@ export function setSelection(
     });
   };
 }
+
+/**
+ * Action dispatched from stepOver action. Receives the position
+ * and depending on where it is paused, sets up hidden breakpoints
+ * and handles resume, stepOver (as needed)
+ * @param position
+ * @returns {function(ThunkArgs)}
+ */
+export function analyzeStepping(type) {
+  return async ({ dispatch, getState, client }: ThunkArgs) => {
+    const position = await dispatch(getPosition());
+    const source = getSelectedSource(getState()).toJS();
+    const hiddenBreakpointLocation = getHiddenBreakpoint(getState());
+
+    await dispatch({
+      type: "ANALYZE_STEPPING",
+      [PROMISE]: (async function() {
+        if (hiddenBreakpointLocation) {
+          await dispatch(removeBreakpoint(hiddenBreakpointLocation));
+        }
+        const path = getClosestPath(source, position);
+        if (!path) {
+          return dispatch(command({ type }));
+        }
+        switch (type) {
+          case "stepOver":
+          case "stepIn":
+          case "stepOut":
+            if (isAwaitExpression(path)) {
+              return dispatch(analyzeAwaitExpression(path, position, type));
+            }
+        }
+        return dispatch(command({ type }));
+      })()
+    });
+  };
+}
```

Comment:
big picture, i'm wondering if we can drop this analyze function in favor of some smaller utilities.

* **getPauseLocationNode(state)** knows how to get the pause location and call getClosest path. 
* **removeHiddenBreakpoint** knows how to check for a hidden bp and remove it on pause.